### PR TITLE
Add more punctuaction highlights and add some missing operators

### PR DESCRIPTION
**Pull request by theHamsta**
"?" is still missing:
https://github.com/tree-sitter/tree-sitter-swift/issues/38

### CODE REVIEW

File: queries/swift/highlights.scm [hunk 1]
```diff
@@ -1,3 +1,6 @@
+[ "." ";" ":" "," ] @punctuation.delimiter
+[ "\\(" "(" ")" "[" "]" "{" "}"] @punctuation.bracket
```
Comment (theHamsta) [inline queries/swift/highlights.scm:2]: ideally "\\("` should be `@punctuation.special` in string interpolation. but it's not easy to match against

`"?"` after optional types needs a parser change: https://github.com/alex-pinkus/tree-sitter-swift/issues/100

File: queries/swift/highlights.scm [hunk 2]
```diff
@@ -1,3 +1,6 @@
+[ "." ";" ":" "," "->" ] @punctuation.delimiter
```
Comment (theHamsta) [inline queries/swift/highlights.scm:1]: `->` is apparently used to indicate return type in functions. `->` is an operator in Rust. So let's switch to that variant

File: tests/query/highlights/swift/string-interpolation.swift [hunk 3]
```diff
@@ -0,0 +1,7 @@
+var one = 1
+var two = 2
+var name = "Let's do some math: \(one) + \(one) = \(one + one) = \(two)"
+//                              ^ <- punctuaction.bracket
```
Comment (clason) [inline tests/query/highlights/swift/string-interpolation.swift:4]:
```suggestion
//                              ^ <- punctuation.bracket
```
(and below)

**Review by clason — APPROVED** [review]

File: tests/query/highlights/swift/string-interpolation.swift [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+//                                 ^ <- variable
+//                                   ^ <- punctuaction.bracket
+//                                                 ^ <- punctuaction.bracket
```
Comment (theHamsta) [inline tests/query/highlights/swift/string-interpolation.swift:7]: `CUSTOM_OPERATOR` as using in `for x in 0..<3` can not be highlighted at the moment

Comment (theHamsta) [reply tests/query/highlights/swift/string-interpolation.swift:7]: `..<` can in some cases.

**Comment by theHamsta** [comment]
I don't understand why tests are failing...

**Comment by theHamsta** [comment]
I will merge without the tests and try to push them in a follow PR!

**Comment by alex-pinkus** [comment]
@theHamsta, any reason I shouldn't upstream these queries into https://github.com/alex-pinkus/tree-sitter-swift? I figure the consistency would be valuable, but want to check in case there's something I'm missing.

**Comment by clason** [comment]
In general, queries are not 100% compatible with upstream since we use some custom/adapted capture names, so we always have to bundle ours. But if you checked them and they apply (with possible minimal changes) upstream, feel free to take them!

**Comment by theHamsta** [comment]
@alex-pinkus i think in this case all changes here could also be made to the upstream queries. In general, the captures Atom expects are a bit different.

**Pull request merged**